src/main/java/com/odysseusinc/arachne/datanode/service/impl/AtlasServiceImpl.java

```diff
@@ -0,0 +1,157 @@
+
+import com.odysseusinc.arachne.datanode.dto.atlas.CohortDefinition;
+import com.odysseusinc.arachne.datanode.exception.ServiceNotAvailableException;
+import com.odysseusinc.arachne.datanode.service.client.atlas.AtlasAuthRequestInterceptor;
+
+
+import com.odysseusinc.arachne.datanode.Constants;
+import com.odysseusinc.arachne.datanode.exception.AuthException;
+import com.odysseusinc.arachne.datanode.service.AtlasService;
+import com.odysseusinc.arachne.datanode.service.client.atlas.AtlasAuthSchema;
+import com.odysseusinc.arachne.datanode.service.client.atlas.AtlasClient;
+import com.odysseusinc.arachne.system.settings.api.v1.dto.SystemSettingDTO;
+import com.odysseusinc.arachne.system.settings.api.v1.dto.SystemSettingsGroupDTO;
+import com.odysseusinc.arachne.system.settings.repository.SystemSettingsGroupRepository;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.core.convert.support.GenericConversionService;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+public class AtlasServiceImpl implements AtlasService {
+    private final RestTemplate atlasRestTemplate;
+    private final GenericConversionService conversionService;
+    private final SystemSettingsGroupRepository systemSettingsGroupRepository;
+    private HttpHeaders headers;
+
+    @Autowired
+    public AtlasServiceImpl(GenericConversionService genericConversionService,
+                            SystemSettingsGroupRepository settingsGroupRepository,
+                            @Qualifier("atlasRestTemplate") RestTemplate atlasRestTemplate) {
+
+        this.conversionService = genericConversionService;
+        this.systemSettingsGroupRepository = settingsGroupRepository;
+        this.atlasRestTemplate = atlasRestTemplate;
+    }
+
+
+    @Override
+    public String checkConnection() {
+
+        SystemSettingsGroupDTO systemSettingsGroup =
+                conversionService.convert(systemSettingsGroupRepository.findByName("atlas"), SystemSettingsGroupDTO.class);
+        Map<String, String> settings = systemSettingsGroup.getFieldList().stream().collect(Collectors.toMap(SystemSettingDTO::getName,
+                SystemSettingDTO::getValue));
+        String url = settings.get("atlas.host") + ":" + settings.get("atlas.port")
```

Comment:
I guess, this was copied from somewhere. Pls, move to some util method and re-use